http_server.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import socket
+import email.utils
+import mimetypes
+
+
+def http_server():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(10)
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+            buffsize = 16
+            msg = ""
+            done = False
+            while not done:
+                part = conn.recv(buffsize)
+                if len(part) < buffsize:
+                    done = True
+                msg = b"{}{}".format(msg, part)
+            out = parse_request(msg)
+            conn.sendall(out)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def response_ok(uri):
+    date = email.utils.formatdate(usegmt=True)
+    code = "HTTP/1.1 200 OK"
+    guess = mimetypes.guess_type(uri)[0]
+    content_type = "Content-Type: {}; charset=UTF-8".format(guess)
+    headers = "{}\r\n{}\r\n{}\r\n\r\n".format(code, date, content_type)
+    response = "{}{}\r\n".format(headers, uri)
+    return response
+
+
+def response_error(err, msg):
+    code = "HTTP/1.1 {} {}".format(err, msg)
+    date = email.utils.formatdate(usegmt=True)
+    content_type = "Content-Type: text/html; charset=UTF-8"
+    response = "{}\r\n{}\r\n{}\r\n\r\n".format(code, date, content_type)
```

Comment:
this same comment applies to `response_ok` above.  The value you have for `date` here is appropriate as a header *value*, but you still need a header name to go with it.  What's the name of the [HTTP header](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) that holds this value?